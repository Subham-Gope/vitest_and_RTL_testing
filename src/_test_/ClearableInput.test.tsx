// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import ClearableInput from "../components/ClearableInput";

// describe("Testing the ClearableInpt.tsx file", () => {
//   test("let's user clear the input", async () => {
//     const { container } = render(<ClearableInput />);

//     expect(container).toMatchSnapshot();

//     const input = screen.getByPlaceholderText("Enter Email");
//     const btn = screen.getByRole("button", { name: /clear/i });

//     expect(input).toBeInTheDocument();
//     expect(btn).toBeInTheDocument();

//     await userEvent.type(input, "xyz@gmail.com");
//     expect(input).toHaveValue("xyz@gmail.com");
//     await userEvent.click(btn);
//     expect(input).toHaveValue("");
//   });
// });
