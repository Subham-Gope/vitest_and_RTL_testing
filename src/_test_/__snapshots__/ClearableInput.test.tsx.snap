// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the ClearableInpt.tsx file > let's user clear the input 1`] = `
<div>
  <div>
    <input
      placeholder="Enter Email"
      type="text"
      value=""
    />
    <button>
      Clear
    </button>
  </div>
</div>
`;
